### ISSUE TITLE
Neither Jenkins core nor Jenkins war have been defined by configuration file or BOM/POM

### ISSUE THREAD

**Issue by ghost** [jenkinsci/custom-war-packager#94]
It seems that the `bom` option in the `packager-config.yml` does not recognize that Jenkins `core` is specified in `bom.yml`.

Steps to reproduce:

**bom.yml**
```
metadata:
  labels:
    name: bom-it
    groupId: org.jenkins-ci.main
    artifactId: jenkins-war
    version: 2.176
spec:
  core:
    version: 2.176
  plugins:
    - groupId: "org.jenkins-ci.plugins.workflow"
      artifactId: "workflow-aggregator"
      version: 2.6
```

**packager-config.yml**
```
bundle:
  groupId: "test.group"
  artifactId: "custom-jenkins"

bomIncludeWar: true
buildSettings:
  bom: bom.yml
```

Run:
```
java -jar custom-war-packager-cli.jar -configPath=packager-config.yml -version=1.0-SNAPSHOT
```

Output:
```
Exception in thread "main" java.io.IOException: Neither Jenkins core nor Jenkins war have been defined by configuration file or BOM/POM
	at io.jenkins.tools.warpackager.lib.impl.Builder.build(Builder.java:106)
	at io.jenkins.tools.warpackager.cli.Main.main(Main.java:53)
```

Adding a valid `war` section to `packager-config.yml` results in a successful build, but all plugins from `bom.yml` are missing.

**packager-config.yml** - modified
```
bundle:
  groupId: "test.group"
  artifactId: "custom-jenkins"

bomIncludeWar: true
buildSettings:
  bom: bom.yml

war:
  groupId: org.jenkins-ci.main
  artifactId: jenkins-war
  source:  
    version: 2.176
```

Output:
```
May 07, 2019 8:45:08 AM io.jenkins.tools.warpackager.lib.impl.Builder buildIfNeeded
INFO: Component org.jenkins-ci.main:jenkins-war:2.176: no build required
```
Things I tried:
https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/20
https://github.com/jenkinsci/jep/tree/master/jep/309

**Workaround**
Add `--bomPath` to the packaging command.
```
java -jar custom-war-packager-cli.jar -configPath=packager-config.yml -version=1.0-SNAPSHOT --bomPath bom.yml
```

**Comment by oleg-nenashev** [comment]
https://github.com/jenkinsci/custom-war-packager/pull/104 should address it

**Comment by oleg-nenashev** [comment]
Finally it was released in 1.7 https://github.com/jenkinsci/custom-war-packager/releases/tag/parent-pom-1.7
